zarr/util.py

```diff
@@ -33,10 +33,23 @@ def flatten(arg: Iterable) -> Iterable:
 }
 
 
+class NumpyEncoder(json.JSONEncoder):
+
+    def default(self, o):
+        # See json.JSONEncoder.default docstring for explanation
+        if isinstance(o, np.integer):
+            return int(o)
+        if isinstance(o, np.floating):
+            return float(o)
```

Comment:
I wonder if we can use `numbers` for this instead. Would need to `import numbers` up top

```suggestion
        if isinstance(o, numbers.Integral):
            return int(o)
        if isinstance(o, numbers.Real):
            return float(o)
```

Reply:
Good point, there is no reason to be numpy specific here.

Reply:
Interesting (worrisome?) that the `numbers.Real` line is not being picked up by codecov.

Reply:
Well I think we still need tests to cover this

Reply:
Changed the test to use `to.float32` instead `np.float64`, because it was hitting a different code path, which is already working fine.